sdk1/src/main/java/com/amazonaws/services/dynamodbv2/datamodeling/encryption/DynamoDBEncryptor.java

```diff
@@ -237,6 +237,16 @@ public static DynamoDBEncryptor getInstance(EncryptionMaterialsProvider provider
     if (attributeFlags.isEmpty()) {
       return itemAttributes;
     }
+    if (!isDecryptionRequired(itemAttributes.keySet(), attributeFlags)) {
+      return itemAttributes;
+    }
+    if (!itemAttributes.containsKey(signatureFieldName)
+        && !itemAttributes.containsKey(materialDescriptionFieldName)) {
```

Comment:
Yes, earlier I tried it but changing to `||` will change existing behaviour. As currently, if any one of the field is missing in that case we throw `signatureException` , as show in [signedOnlyNoSignature](https://github.com/aws/aws-dynamodb-encryption-java/blob/873b7a725f4df7c0fea0a337fffea8ba82bb717f/sdk1/src/test/java/com/amazonaws/services/dynamodbv2/datamodeling/encryption/DynamoDBEncryptorTest.java#L297) test.

Not sure if such a change will be considered backward-incompatible, If any client is checking for `signatureException` and doing some logic it will break. Hence to not change the existing behaviour i think `&&` is better option.

Reply:
make sense done